File: frontend/src/lib/mshnctrl/api/roe.ts

import { api } from '@/lib/api';

export interface Roe {
    id: string;
    name: string;
    description: string;
    status: 'ACTIVE' | 'CANCELLED' | 'DRAFT';
    roe_type: 'SELF_DEFENSE' | 'MISSION_SPECIFIC' | 'STANDING';
    created_at: string;
    created_by: string;
}

export interface CreateRoeRequest {
    name: string;
    description: string;
    roe_type: 'SELF_DEFENSE' | 'MISSION_SPECIFIC' | 'STANDING';
    status: 'ACTIVE' | 'CANCELLED' | 'DRAFT';
    created_by: string;
}

export interface RoeComplianceRequest {
    target_id: string;
    proposed_action: string;
}

export interface RoeComplianceResult {
    compliant: boolean;
    violated_rules: string[];
    caveats: string[];
}

export const RoeApi = {
    getActiveRoes: async (): Promise<Roe[]> => {
        return api.get<Roe[]>('/roe');
    },

    createRoe: async (req: CreateRoeRequest): Promise<{ id: string }> => {
        return api.post<{ id: string }>('/roe', req);
    },

    checkCompliance: async (req: RoeComplianceRequest): Promise<RoeComplianceResult> => {
        return api.post<RoeComplianceResult>('/roe/check', req);
    }
};

File: backend/src/features/targeting/domain/roe.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Roe {
    pub id: String,
    pub name: String,
    pub description: String,
    pub status: String, // ACTIVE, CANCELLED, DRAFT
    pub roe_type: String, // SELF_DEFENSE, MISSION_SPECIFIC, STANDING
    pub created_at: String,
    pub created_by: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateRoeRequest {
    pub name: String,
    pub description: String,
    pub roe_type: String, // SELF_DEFENSE, MISSION_SPECIFIC, STANDING
    pub status: String,   // Defaults to ACTIVE if omitted likely
    pub created_by: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RoeComplianceRequest {
    pub target_id: String,
    pub proposed_action: String, // e.g., "KINETIC_STRIKE", "CAPTURE"
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RoeComplianceResult {
    pub compliant: bool,
    pub violated_rules: Vec<String>,
    pub caveats: Vec<String>,
}
